### PR TITLE
Fixes GDPR bug about being overly strict on publisher restrictions

### LINKED ISSUE #1709: cookie sync ok but no buyeruidsent to bidder
Hi,

I have a strange behavior with one specific publisher for one specific bidder. On this publisher, buyeruid is never sent to bidder rubicon even though endpoint cookie_sync was called right before to check cookie for this bidder and sync url was sent. I checked the consent string and rubicon's vendor id was allowed.
With other publishers I don't have this issue and buyerid is sent in the payload to rubicon

So my question would be : what could avoid buyeruid to be sent to bidder even thought it's in the uids cookie ? I suspect something related with the consent string but it looks ok to me. For instance, here one of the consent string where this issue happens :

CPAavcCPAavcCAGABCFRBKCsAP_AAH_AAAqIHFNf_X_fb3_j-_59_9t0eY1f9_7_v-0zjgeds-8Nyd_X_L8X5mM7vB36pq4KuR4Eu3LBAQdlHOHcTUmw6IkVqTPsbk2Mr7NKJ7PEinMbe2dYGH9_n9XT_ZKY79_____7__-_____7_f__-__3_vp9V---wOJAIMBAUAgAEMAAQIFCIQAAQhiQAAAABBCIBQJIAEqgAWVwEdoIEACAxAQgQAgBBQgwCAAQAAJKAgBACwQCAAiAQAAgAEAIAAEIAILACQEAAAEAJCAAiACECAgiAAg5DAgIgCCAFABAAAuJDACAMooASBAPGQGAAKAAqACGAEwALgAjgBlgDUAHZAPsA_ACMAFLAK2AbwBMQCbAFogLYAYEAw8BkQDOQGeAM-EQHwAVABWAC4AIYAZAAywBqADZAHYAPwAgABGAClgFPANYAdUA-QCGwEOgIvASIAmwBOwCkQFyAMCAYSAw8Bk4DOQGfCQAYADgBzgN_CQTgAEAALgAoACoAGQAOAAeABAACIAFQAMIAaABqADyAIYAigBMgCqAKwAWAAuABvADmAHoAQ0AiACJgEsAS4AmgBSgC3AGGAMgAZcA1ADVAGyAO8AewA-IB9gH6AQAAjABQQClgFPAL8AYoA1gBtADcAG8AOIAegA-QCGwEOgIqAReAkQBMQCZQE2AJ2AUOApEBYoC2AFyALvAYEAwYBhIDDQGHgMiAZIAycBlwDOQGfANIAadA1gDWQoAEAYQaBIACoAKwAXABDADIAGWANQAbIA7AB-AEAAIKARgApYBT4C0ALSAawA3gB1QD5AIbAQ6Ai8BIgCbAE7AKRAXIAwIBhIDDwGMAMnAZyAzwBnwcAEAA4Bv4qA2ABQAFQAQwAmABcAEcAMsAagA7AB-AEYAKXAWgBaQDeAJBATEAmwBTYC2AFyAMCAYeAyIBnIDPAGfANyHQWQAFwAUABUADIAHAAQAAiABdADAAMYAaABqADwAH0AQwBFACZAFUAVgAsABcADEAGYAN4AcwA9ACGAERAJYAmABNACjAFKALEAW4AwwBkADKAGiANQAbIA3wB3gD2gH2AfoBGACVAFBAKeAWKAtAC0gFzALyAX4AxQBuADiQHTAdQA9ACGwEOgIiAReAkEBIgCbAE7AKHAU0AqwBYsC2ALZAXAAuQBdoC7wGEgMNAYeAxIBjADHgGSAMnAZUAywBlwDOQGfANEgaQBpIDSwGnANYAbGPABAIqAb-QgZgALAAoABkAEQALgAYgBDACYAFUALgAYgAzABvAD0AI4AWIAygBqADfAHfAPsA_ACMAFBAKGAU-AtAC0gF-AMUAdQA9ACQQEiAJsAU0AsUBaMC2ALaAXAAuQBdoDDwGJAMiAZOAzkBngDPgGiANJAaWA4AlAyAAQAAsACgAGQAOAAigBgAGIAPAAiABMACqAFwAMQAZgA2gCGgEQARIAowBSgC3AGEAMoAaoA2QB3gD8AIwAU-AtAC0gGKANwAcQA6gCHQEXgJEATYAsUBbAC7QGHgMiAZOAywBnIDPAGfANIAawA4AmACARUA38pBBAAXABQAFQAMgAcABAACKAGAAYwA0ADUAHkAQwBFACYAFIAKoAWAAuABiADMAHMAQwAiABRgClAFiALcAZQA0QBqgDZAHfAPsA_ACMAFBAKGAVsAuYBeQDaAG4APQAh0BF4CRAE2AJ2AUOApoBWwCxQFsALgAXIAu0BhoDDwGMAMiAZIAycBlwDOQGeAM-gaQBpMDWANZAbGVABAA-Ab-A.YAAAAAAAAAAA

### PR DESCRIPTION
**Pull request by hhhjort**
We were requiring the vendor primary purpose match the restricted purpose in flex cases, not that we were only looking at the purpose/legit interest when restricted to do so.

Fixes #1709

### CODE REVIEW

File: gdpr/impl_test.go [hunk 1]
```diff
@@ -389,6 +397,14 @@ func TestAllowPersonalInfoTCF2(t *testing.T) {
...
+			allowPI:     true,
+			allowGeo:    false,
+			allowID:     true,
```
Comment (SyntaxNode) [inline gdpr/impl_test.go:406]: They key to this test is to have a publisher restriction for a purpose which requires either consent or legitimate interest that applies to a vendor that declares "flex" for that purpose and has the opposite as primary. So, if they are restricting to consent the vendor must have legitimate interest as primary on the "flex".

@hhhjort Could you please add comment above the description to more specifically detail the publisher restriction and I'll try to create a shorter consent string for this test.

Comment (SyntaxNode) [reply gdpr/impl_test.go:406]: Please also mention which vendors this is relying on.

Comment (SyntaxNode) [reply gdpr/impl_test.go:409]: I'll update the consent string in the future. I don't have the time to today and don't want that to hold up the fix.

**Review by SyntaxNode — APPROVED** [review]

**Review by bsardo — APPROVED** [review]
LGTM

**Pull request merged**